### PR TITLE
DSND-2369: Implement handling of NEWINC with embedded SH01

### PR DESCRIPTION
**Pull request by cgregoryCH**
## Describe the changes
*  Implement handling of NEWINC with embedded SH01 associated filing.

### Related Jira tickets

[DSND-2369](https://companieshouse.atlassian.net/browse/DSND-2369)

### General

- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2369]: https://companieshouse.atlassian.net/browse/DSND-2369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java [hunk 1]
```diff
@@ -42,7 +49,50 @@ public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(Inter
...
+
+            Optional.ofNullable(mappedData.getAssociatedFilings())
+                    .ifPresentOrElse(
+                            associatedFilingList -> associatedFilingList.stream()
+                                    .filter(associatedFiling -> requestEntityId.equals(associatedFiling.getEntityId()))
+                                    .findFirst()
+                                    .ifPresentOrElse(associatedFiling -> {
+                                                if (isDeltaStale(request.getInternalData().getDeltaAt(),
+                                                        associatedFiling.getDeltaAt())) {
+                                                    LOGGER.error(STALE_DELTA_ERROR_MESSAGE.formatted(
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java:66]: It looks like there is missing test coverage for this block

File: src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java [hunk 2]
```diff
@@ -42,7 +49,50 @@ public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(Inter
...
+                                                // Update already existing associated filing from existing list
+                                                associatedFilingChildMapper.mapChild(associatedFiling, request);
+                                            },
+                                            // Add new associated filing to existing list
+                                            () -> {
+                                                if (associatedFilingList.stream()
+                                                        .anyMatch(associatedFiling ->
+                                                                StringUtils.isBlank(associatedFiling.getEntityId()))) {
+                                                    LOGGER.info(
+                                                            MISSING_ENTITY_ID_ERROR_MSG.formatted(requestEntityId),
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java:82]: It looks like there is missing test coverage for this block

File: src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java [hunk 3]
```diff
@@ -42,7 +49,50 @@ public FilingHistoryDocument mapFilingHistoryToExistingDocumentUnlessStale(Inter
...
+                                                            DataMapHolder.getLogMap()
+                                                    );
+                                                }
+                                                associatedFilingList
+                                                        .add(associatedFilingChildMapper
```
Comment (johncook-ch) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/mapper/upsert/TopLevelTransactionMapper.java:87]: It looks like there is missing test coverage for this block

**Review by johncook-ch — COMMENTED** [review]
Code coverage for `TopLevelTransactionMapper` is < 80% in IntelliJ

**Comment by sthompsonCH** [comment]
Missing test coverage will be added as part of a separate PR for this Jira ticket

**Review by johncook-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Pull request merged**